File: modules/api/routes/loadouts.py
"""装配清单（loadout，B1）：`runtime/loadouts/<id>.yaml` 的只读清单。

loadout = 三件套引用（map_plan / strategy / plan[+spawn]），`POST /api/session/start?loadout=`
解析它并自动把生产序列 submit 入队。没有写面 —— 文件由人管（与锁定预设同一姿态）。
"""
from __future__ import annotations

from fastapi import APIRouter, Request

router = APIRouter()


@router.get("/api/loadouts")
def loadouts_list(request: Request) -> list[dict]:
    """清单（StartCard 的下拉与 `start_session(loadout=)` 都吃它）。"""
    return request.app.state.loadouts.list()

File: web/src/api/loadouts.ts
/**
 * 装配清单（loadout，B1）：三件套引用（map_plan / strategy / plan[+spawn]）。
 * 只读清单 —— 文件由人管（runtime/loadouts/<id>.yaml），没有写面。
 */
import { API_BASE } from "../store/frames";

export interface LoadoutMeta {
  id: string;
  title_zh: string;
  map_plan: string | null;
  strategy: string | null;
  plan: string | null;
  spawn: string | null;
}

export async function listLoadouts(): Promise<LoadoutMeta[]> {
  const res = await fetch(new URL("/api/loadouts", API_BASE));
  if (!res.ok) throw new Error(`后端返回 ${res.status}`);
  return (await res.json()) as LoadoutMeta[];
}
